src/service/video/docking.js

```diff
@@ -601,53 +629,89 @@ export class VideoDocking {
   }
 
   /**
-   * Reconciliates the state of a docked or potentially dockable video when
+   * Reconciles the state of a docked or potentially dockable video when
    * the viewport/position changes.
    * @param {!../../video-interface.VideoOrBaseElementDef} video
-   * @return {{posX: RelativeX, posY: RelativeY}|undefined}
+   * @return {?DockTargetDef}
    * @private
    */
-  update_(video) {
-    if (this.isDragging_) {
-      return;
-    }
-    if (this.ignoreDueToSize_(video) ||
+  getTargetFor_(video) {
+    if (this.isDragging_ ||
+        this.ignoreDueToSize_(video) ||
         this.ignoreBecauseAnotherDocked_(video) ||
-        this.ignoreDueToNotPlayingManually_(video)) {
-      return;
+        this.ignoreDueToNotPlayingManually_(video) ||
+        this.undockBecauseVisible_(video)) {
```

Comment:
It looks like `undockBecauseVisible_` has side effects, and calling methods with side effects inside `if` conditions can add complexity to the logical flow of the code. Consider refactoring

Reply:
Coming in a follow-up PR.